### DISCUSSION TITLE
Enable native ssh feature for linux machines

### DISCUSSION THREAD

**Discussion by slax81** [Ylianst/MeshCentral#4867] [Q&A] [answered]
Hello, I need help enabling SSH from meshcentral terminal for Linux machines without using an agent on that machine. I am under impression that this is possible, however I can't get it to work. When I add linux host, I have only VNC and RDP options (links). I run the latest 1.1.1 version as docker image in kubernetes. I also saw some online info that SSH should be enabled in config.json so I changed it, but maybe I'm missing some more options?
`/meshcentral/meshcentral-data $ cat ./config.json 
{
  "settings": {
    "ssh": true
  }
}`

**Comment by slax81** [comment]
/meshcentral/meshcentral-data $ cat ./config.json 
{
  "settings": {
    "ssh": true
  }
}

**Reply by si458** [reply]
The ssh feature I believe it only possible using the relay feature

The meshserver connects to a device running the meshagent then that device connects to your remote device you want

You also have to create a standard group and not an meshagent group I believe too

**Reply by slax81** [reply]
Yes, the group is 	Local devices, no agent
And this guy here says it's working for him:
`https://www.reddit.com/r/MeshCentral/comments/xu1iqr/cannot_ssh_to_linux_systems/`

**Reply by slax81** [reply]
@si458 If I understand correctly, I would need a dedicated Linux machine as relay to have an SSH feature? Is there any doc to read on how to set up that?

**Reply by si458** [reply]
@Ylianst has loads of youtube videos, one of them will help you - https://www.youtube.com/@MeshCentral/videos

EDIT: https://www.youtube.com/watch?v=CHZYFduiOx0 this video is about local device support without an agent, give it a watch

**Comment by slax81** [comment, marked as answer]
The config file was not written correctly and that was the issue. This is the working config:

{
  "settings": {}, 
  "domains": {
    "": {
      "title": "title",
      "title2": "title2",
      "mstsc": true,
      "ssh": true
    }
  }
}
